business-networks-workflows/src/main/kotlin/net/corda/bn/flows/MembershipManagementExceptions.kt

```diff
@@ -63,4 +63,9 @@ class IllegalFlowArgumentException(message: String) : FlowException(message)
 /**
  * Exception thrown by any [MembershipManagementFlow] whenever group ends up in illegal state.
  */
-class IllegalBusinessNetworkGroupStateException(message: String) : FlowException(message)
\ No newline at end of file
+class IllegalBusinessNetworkGroupStateException(message: String) : FlowException(message)
+
+/**
+ * Exception thrown by any [MembershipManagementFlow] whenever the network is left in an invalid state(i.e. no members with permission to revoke memberships)
+ */
+class InvalidBusinessNetworkStateException(message: String) : FlowException(message)
```

Comment:
I don't agree here. Illegal and invalid have their own place. Illegal can mean that the state can still be valid but it just not OK because of arbitrary rules. Invalid simply means the state is in such a poor state the network cannot function properly or even at all.